src/audio/mux/mux.c

```diff
@@ -458,9 +458,12 @@ static int demux_process(struct processing_module *mod,
 
```

Comment:
a general comment to this commit: this should work now too, it shouldn't be the case, that "This doesn't
work in the tree currently." But by looking at the code it seems to me now, that commit https://github.com/thesofproject/sof/commit/c399624fcfb7dfaf75d69671bc2a91ec9812a6ad broke it. Before that commit we had checks like https://github.com/thesofproject/sof/blob/bca7bf436e8ee9d24cbf244ff78b8bd034cc2897/src/audio/mux/mux.c#L673 to only write to active pipelines, after that commit we now seem to write to all sinks https://github.com/thesofproject/sof/blob/c399624fcfb7dfaf75d69671bc2a91ec9812a6ad/src/audio/mux/mux.c#L511-L516 although previously we only initialised some elements of the `sinks_stream[]` array in https://github.com/thesofproject/sof/blob/c399624fcfb7dfaf75d69671bc2a91ec9812a6ad/src/audio/mux/mux.c#L487-L497 Am I right? Is that the bug, that you're fixing here? Then maybe we just should fix that bug?

Reply:
That's pretty much the form of this patch indeed; obviously I didn't realize it was fixing a regression.  Will dig up the history and annotate the commit message.

FWIW, the other half of this is the overrun flag, which is new and needed.  Stream startup is async and under host control, at least on the board I'm looking at it takes ~20-30ms between the pipeline start IPC and the first mic bytes to arrive via DMA.    If we allow the demux to write once the sink component goes active, it hits an unavoidable xrun condition (the handling for which seems... fragile and untested?  Once that happens the firmware/kernel state is borked and I need to reload it all[1].

[1] Which requires a reboot on mt8195 because the kernel panics if I try to unload the module!